addon/-private/validated-changeset.ts

```diff
@@ -157,13 +157,9 @@ export class BufferedChangeset implements IChangeset {
     }
 
     return keys(obj).map(key => {
-      let value = transform(obj[key]);
+      let { value, validation } = transform(obj[key]);
 
-      if (isObject(value)) {
```

Comment:
This was always true as it as always returned an object from the transform method.